src/zos-mkdtemp.c

```diff
@@ -0,0 +1,105 @@
+//////////////////////////////////////////////////////////////////////////////
+//// Licensed Materials - Property of IBM
+//// ZOSLIB
+//// (C) Copyright IBM Corp. 2020. All Rights Reserved.
+//// US Government Users Restricted Rights - Use, duplication
+//// or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+/////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/stat.h>
+
+typedef enum {
+  cp_ebcdic,
+  cp_ascii
+} CodePage;
+
+#define XXXXXX "XXXXXX"
+#define ValidChars "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_.,-"
+typedef struct CodePage {
+  char xes[sizeof(XXXXXX)];
+  char validChars[sizeof(ValidChars)];
+}  CodePage_t;
+
+#pragma convert("IBM-1047")
+static const CodePage_t CharTraitsEbcdic = {
+   XXXXXX,
+   ValidChars
+};
+#pragma convert(pop)
+
+#pragma convert("ISO8859-1")
```

Comment:
pragma convert works in xlclang and clang?

Reply:
yes